From a8f3a593c84bf1523c6e9ba517df01ef2e1fc894 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Tue, 10 Jul 2018 17:38:15 +0300
Subject: [PATCH] board: dali is a new ARM64 SoC

Put preboot inside BTL instead of SPL
This will decrease time to bring up PCIe link
---

diff --git a/board/habanalabs/dali/post-image.sh b/board/habanalabs/dali/post-image.sh
index e8faecc..e42cfa9 100755
--- a/board/habanalabs/dali/post-image.sh
+++ b/board/habanalabs/dali/post-image.sh
@@ -32,25 +32,37 @@
 	${HOST_DIR}/usr/bin/mkimage -f dali_fit_spl.its -E ${UBOOT_FITFILE}
 	${HOST_DIR}/usr/bin/mkimage -C none -A arm64 -T script -d ${UBOOT_CMDFILE} boot.scr
 
-	# KMD likes these to be 4B aligned in size
+	dd status=none conv=sync bs=8 if=${SPL_BINFILE}						 > dali-u-boot-spl.bin
+	# Swap SPL
+	swap_file dali-u-boot-spl.bin dali-u-boot-spl.bin.be
+	SPL_BINFILE=dali-u-boot-spl.bin.be
+
+	# KMD like this to be 4B aligned in size
 	dd status=none conv=sync bs=4 if=${LINUX_FITFILE}					 > dali-fit.itb
+	# Used by TRACE32 (Lauterbach JTAG)
 	dd status=none conv=sync bs=4 if=u-boot.bin						 > dali-u-boot.bin
-
-	# BTL likes these to be 16B aligned in size
-	dd status=none conv=sync bs=16 if=${SPL_BINFILE}	> dali-u-boot-spl.bin
-
-	# prepare BTL image
-	UBOOT_SPI_SIZE=$(printf "0x%X\n" `stat -c "%s" ${UBOOT_BINFILE}`)
-	cp -a ${BOARD_BASE}/btl .
-	make -C btl clean all UBOOT_SPI_SIZE=${UBOOT_SPI_SIZE}
-	cp btl/sectors-be.bin .
 
 	# Swap FW
 	swap_file btl/pcie_phy_fw.bin pcie_phy_fw.bin.be
+	PCIE_PHY_FW_BINFILE=pcie_phy_fw.bin.be
 
 	### Construct bin file
+	# BTL likes these to be 16B aligned in size
+	dd status=none conv=sync bs=16 if=preboot.bin						 > dali-preboot.bin
+	# prepare BTL image
+	BTL_CMD_ARGS="${BTL_CMD_ARGS} SPL_SIZE=$(printf "0x%X\n" `stat -c "%s" ${SPL_BINFILE}`)"
+	BTL_CMD_ARGS="${BTL_CMD_ARGS} UBOOT_SIZE=$(printf "0x%X\n" `stat -c "%s" ${UBOOT_BINFILE}`)"
+	BTL_CMD_ARGS="${BTL_CMD_ARGS} PCIE_PHY_FW_SIZE=$(printf "0x%X\n" `stat -c "%s" ${PCIE_PHY_FW_BINFILE}`)"
+	cp -a ${BOARD_BASE}/btl .
+	make -C btl clean all ${BTL_CMD_ARGS}
+	cp btl/sectors-be.bin .
+
 	# BTL
 	dd status=none conv=sync bs=256K	if=sectors-be.bin				 > dali-spi-flash.bin.nocrc
+	# Primary SPL
+	dd status=none conv=sync bs=64K		if=${SPL_BINFILE}				>> dali-spi-flash.bin.nocrc
+	# Secondary SPL
+	dd status=none conv=sync bs=64K		if=${SPL_BINFILE}				>> dali-spi-flash.bin.nocrc
 	# Primary U-Boot
 	dd status=none conv=sync bs=1M		if=${UBOOT_BINFILE}				>> dali-spi-flash.bin.nocrc
 	# Secondary U-Boot
@@ -60,9 +72,9 @@
 	# U-Boot ENV Redundand
 	dd status=none conv=sync bs=64K		if=uboot-env.bin				>> dali-spi-flash.bin.nocrc
 	# PCIe FW
-	dd status=none conv=sync bs=64K		if=pcie_phy_fw.bin.be				>> dali-spi-flash.bin.nocrc
+	dd status=none conv=sync bs=64K		if=${PCIE_PHY_FW_BINFILE}			>> dali-spi-flash.bin.nocrc
 	# PCIe FW Redundand
-	dd status=none conv=sync bs=64K		if=pcie_phy_fw.bin.be				>> dali-spi-flash.bin.nocrc
+	dd status=none conv=sync bs=64K		if=${PCIE_PHY_FW_BINFILE}			>> dali-spi-flash.bin.nocrc
 
 	# Prepare checksum
 	CKSUM=`cksum dali-spi-flash.bin.nocrc | cut -d' ' -f1`
